js/modules/k6/ws/ws_test.go

```diff
@@ -1288,3 +1289,40 @@ func TestCookieJar(t *testing.T) {
 
 	assertSessionMetricsEmitted(t, metrics.GetBufferedSamples(ts.samples), "", sr("WSBIN_URL/ws-echo-someheader"), statusProtocolSwitch, "")
 }
+
+func TestWithoutThrowError(t *testing.T) {
+	tb := httpmultibin.NewHTTPMultiBin(t)
+	root, err := lib.NewGroup("", nil)
+	require.NoError(t, err)
+
+	rt := goja.New()
+	rt.SetFieldNameMapper(common.FieldNameMapper{})
+	samples := make(chan metrics.SampleContainer, 1000)
+	state := &lib.State{
+		Group:  root,
+		Dialer: tb.Dialer,
+		Options: lib.Options{
+			SystemTags: &metrics.DefaultSystemTagSet,
+			Throw:      null.BoolFrom(false),
+		},
+		Samples:        samples,
+		BuiltinMetrics: metrics.RegisterBuiltinMetrics(metrics.NewRegistry()),
+		Tags:           lib.NewTagMap(nil),
+	}
+
+	m := New().NewModuleInstance(&modulestest.VU{
+		CtxField:     context.Background(),
+		InitEnvField: &common.InitEnvironment{},
+		RuntimeField: rt,
+		StateField:   state,
+	})
+	require.NoError(t, rt.Set("ws", m.Exports().Default))
+	_, err = rt.RunString(`
+		var res = ws.connect("INVALID", function(socket){
+			socket.on("open", function() {
+				socket.close();
+			});
+		});
```

Comment:
You should likely check that the response is non `null` and has an `error` in it.